File: Data Structures & Algorithms/is-anagram/submission-3.java

class Solution {
    public boolean isAnagram(String s, String t) {
        int n=s.length();
        int m=t.length();
        if (n!=m ) {
            return false;
        }
        int [] arr = new int[26];
        for(int i=0; i<s.length() ;i++ ) {
            arr[s.charAt(i) - 'a']++;
            arr[t.charAt(i) - 'a']--;
        }
        
        for (int x : arr) {
            if (x != 0) return false;
        }
        return true;
    }
}

File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-3.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        curr_max=arr[-1]
        
        for i in range(len(arr)-2,-1,-1) :
            tmp=curr_max
            if arr[i]>curr_max :
                tmp=arr[i]
            arr[i]=curr_max
            curr_max=tmp
        arr[-1]=-1
        return arr


        